orttraining/orttraining/python/training/orttrainer.py

```diff
@@ -843,3 +854,306 @@ def _update_onnx_model_initializers(self, state_tensors):
+
+        def _check_key_mismatch(training_session_state_dict, state_dict):
+            model = 'model'
+            optimizer = 'optimizer'
+            # check presence of 'model' in the input state_dict
+            if model not in state_dict:
+                raise RuntimeError("Missing key: model in state_dict")
+            # check presence of 'optimizer' in the input state_dict
+            if optimizer not in state_dict:
+                raise RuntimeError("Missing key: optimizer in state_dict")
+
+            _check_model_key_mismatch(training_session_state_dict, state_dict)
+
+            _check_optimizer_key_mismatch(training_session_state_dict, state_dict)
+
+
+        # extract state dict from the current training session. this is to persist the states between
+        # two training sessions.
+        # for example, if user provided only the model states, the optimizer states from the current
+        # training session must be persisted
+        states = {}
+        if self._training_session:
+            states = self.state_dict()
+            if strict:
+                _check_key_mismatch(states, state_dict)
+
+        # load the model states from the input state dictionary into the onnx graph and update the
+        # training session states dictionary
+        self._load_model_states(states, state_dict, strict)
+
+        # load the optimizer states from the input state dictionary into the training session states
+        # dictionary
+        self._load_optimizer_states(states, state_dict)
+
+        return states
+
+    def load_state_dict(self, state_dict, strict=True):
+        """Loads input state dictionary containing model and optimizer states into the training session.
+
+        - Load the model states from input state dictionary
+        - Load the optimizer states from input state dictionary
+        - If onnx graph has not been initialized, the states will not be immediately loaded onto the graph;
+        the states will be loaded once the training session has been initialized when either train_step or eval_step is called
+
+        Args:
+            state_dict: state dictionary containing both model and optimizer states. The structure of this dictionary
+                should be the same as the one that is returned by ORTTrainer.state_dict for the case when pytorch_format=False
+            strict: boolean flag to strictly enforce that the input state_dict keys match the keys from ORTTrainer.state_dict
+        """
+
+        # if onnx graph has not been initialized, loading of states will be put on hold.
+        # a copy of the state_dict and other arguments to the function will be stored until the onnx graph has
+        # been initialized. Once the graph is initialized, the desired states will be loaded onto the grpah
+        if not self._training_session:
+            self._load_state_dict = partial(self._load_state_dict_impl, state_dict, strict=strict)
```

Comment:
Why are we storing a method call instead of user state_dict here? This is a change from current implementation

Reply:
This way I can package the arguments together in a single callable. It is a slight modification of our existing implementation, but it enables us not to store ```state_dict``` and ```strict``` separately.